WordPress/Classes/ViewRelated/Reader/ReaderCrossPostCell.swift

```diff
@@ -1,211 +1,142 @@
-        imageView.layer.borderWidth = Constants.imageBorderWidth
-        imageView.layer.masksToBounds = true
-    }
+    func configure(with post: ReaderPost) {
+        headerLabel.attributedText = makeHeaderString(for: post)
+        postTitleLabel.attributedText = NSAttributedString(string: post.titleForDisplay() ?? "", attributes: postTitleAttributes)
 
-    func configureTitleLabel() {
-         if var title = contentProvider?.titleForDisplay(), !title.isEmpty() {
-            if let prefixRange = title.range(of: Constants.xPostTitlePrefix) {
-                title.removeSubrange(prefixRange)
-            }
-
-            titleLabel.attributedText = NSAttributedString(string: title, attributes: readerCrossPostTitleAttributes)
-            titleLabel.isHidden = false
-        } else {
-            titleLabel.attributedText = nil
-            titleLabel.isHidden = true
+        avatarView.setPlaceholder(UIImage(named: "post-blavatar-placeholder"))
+        if let avatarURL = post.avatarURLForDisplay() {
+            let avatarSize = CGSize(width: avatarSize, height: avatarSize)
+                .scaled(by: UITraitCollection.current.displayScale)
+            avatarView.setImage(with: avatarURL, size: avatarSize)
         }
     }
 
-    func configureLabel() {
-        guard let contentProvider = contentProvider else {
-            return
+    private func makeHeaderString(for post: ReaderPost) -> NSAttributedString? {
+        guard let meta = post.crossPostMeta else {
+            return nil
         }
+        let template = meta.commentURL.isEmpty ? Strings.siteTemplate : Strings.commentTemplate
 
-        // Compose the subtitle
-        // These templates are deliberately not localized (for now) given the intended audience.
-        let template = contentProvider.isCommentCrossPost() ? Constants.commentTemplate : Constants.siteTemplate
-
-        let authorName: NSString = contentProvider.authorForDisplay() as NSString
-        let siteName = subDomainNameFromPath(contentProvider.siteURLForDisplay())
-        let originName = subDomainNameFromPath(contentProvider.crossPostOriginSiteURLForDisplay())
+        let authorName: NSString = post.authorForDisplay() as NSString
+        let siteName = subdomainNameFromPath(post.blogURL)
+        let originName = subdomainNameFromPath(meta.siteURL)
 
         let subtitle = NSString(format: template as NSString, authorName, originName, siteName) as String
-        let attrSubtitle = NSMutableAttributedString(string: subtitle, attributes: readerCrossPostSubtitleAttributes)
-
-        attrSubtitle.setAttributes(readerCrossPostBoldSubtitleAttributes, range: NSRange(location: 0, length: authorName.length))
+        let string = NSMutableAttributedString(string: subtitle, attributes: subtitleAttributes)
 
+        string.setAttributes(boldSubtitleAttributes, range: NSRange(location: 0, length: authorName.length))
```

Comment:
The localized template may not start with `authorName`.

I'm aware this is not a new change. Feel free to leave it if you don't think that's an issue.

Reply:
Oh yeah good catch. I opted to revert the localization for now to focus on other priorities.